packages/evm/src/interpreter.ts

```diff
@@ -259,41 +266,54 @@ export class Interpreter {
   async runStep(): Promise<void> {
     const opInfo = this.lookupOpInfo(this._runState.opCode)
 
-    let gas = BigInt(opInfo.fee)
-    // clone the gas limit; call opcodes can add stipend,
-    // which makes it seem like the gas left increases
-    const gasLimitClone = this.getGasLeft()
-
-    if (opInfo.dynamicGas) {
-      const dynamicGasHandler = (this._evm as any)._dynamicGasHandlers.get(this._runState.opCode)!
-      // This function updates the gas in-place.
-      // It needs the base fee, for correct gas limit calculation for the CALL opcodes
-      gas = await dynamicGasHandler(this._runState, gas, this.common)
-    }
+    let timer: Timer
 
-    if (this._evm.events.listenerCount('step') > 0 || this._evm.DEBUG) {
-      // Only run this stepHook function if there is an event listener (e.g. test runner)
-      // or if the vm is running in debug mode (to display opcode debug logs)
-      await this._runStepHook(gas, gasLimitClone)
+    if (this.profilerOpts?.enabled === true) {
+      timer = this.performanceLogger.startTimer(opInfo.name)
     }
 
-    // Check for invalid opcode
-    if (opInfo.name === 'INVALID') {
-      throw new EvmError(ERROR.INVALID_OPCODE)
-    }
+    let gas = BigInt(opInfo.fee)
 
-    // Reduce opcode's base fee
-    this.useGas(gas, `${opInfo.name} fee`)
-    // Advance program counter
-    this._runState.programCounter++
+    try {
+      // clone the gas limit; call opcodes can add stipend,
+      // which makes it seem like the gas left increases
+      const gasLimitClone = this.getGasLeft()
 
-    // Execute opcode handler
-    const opFn = this.getOpHandler(opInfo)
+      if (opInfo.dynamicGas) {
+        const dynamicGasHandler = (this._evm as any)._dynamicGasHandlers.get(this._runState.opCode)!
+        // This function updates the gas in-place.
+        // It needs the base fee, for correct gas limit calculation for the CALL opcodes
+        gas = await dynamicGasHandler(this._runState, gas, this.common)
+      }
 
-    if (opInfo.isAsync) {
-      await (opFn as AsyncOpHandler).apply(null, [this._runState, this.common])
-    } else {
-      opFn.apply(null, [this._runState, this.common])
+      if (this._evm.events.listenerCount('step') > 0 || this._evm.DEBUG) {
+        // Only run this stepHook function if there is an event listener (e.g. test runner)
+        // or if the vm is running in debug mode (to display opcode debug logs)
+        await this._runStepHook(gas, gasLimitClone)
+      }
+
+      // Check for invalid opcode
+      if (opInfo.name === 'INVALID') {
+        throw new EvmError(ERROR.INVALID_OPCODE)
```

Comment:
@acolytec3 see here, this is inside the try/catch

I don't know why this git diff is weird, because I think I only added the try catch blocks and the timers, nothing else, but here it seems like I added a whole bunch of EVM logic (which I did not do)

Reply:
I'm not sure, I don't see all of this as a diff on my local but it looks fine.